linode/design-language-system
# feat: Provide actual values in typescript definition files

## Description 📝
In order to provide better type definitions to developers, we want to provide them with the actual values of the tokens, not just what primitive it is.

## Major Changes 🔄
- Upgraded `sd-transforms` and `style-dictionary`:
   - "@tokens-studio/sd-transforms": "^0.15.2"
   - "style-dictionary": "4.0.0-prerelease.25"
- Using `moduleResolution: "Bundler"` - to allow package entrypoints to work and for those imports to be resolved
- Changed format `typescript/es6-declarations` to `javascript/es6` fixed this for non-nested tokens
- For nested file, returning a string literal instead of the `typeof`. I was able to clean up `formatProperties` significantly.
- We're now using shorthand for typography tokens

## Preview 📷
| Before  | After   |
| ------- | ------- |
| <img width="440" alt="Screenshot 2024-04-12 at 4 59 26 PM" src="https://github.com/linode/design-language-system/assets/125309814/2aca61fd-a0e7-4a07-bb38-acaed44cca56"> | <img width="466" alt="Screenshot 2024-04-12 at 4 59 56 PM" src="https://github.com/linode/design-language-system/assets/125309814/d73cc421-9e6e-470b-a629-e2add5a6101d"> |
|<img width="209" alt="Screenshot 2024-04-12 at 4 59 12 PM" src="https://github.com/linode/design-language-system/assets/125309814/6bf68ca7-c412-44da-bf5d-0f0171787160">|<img width="203" alt="Screenshot 2024-04-12 at 4 56 41 PM" src="https://github.com/linode/design-language-system/assets/125309814/7f5b7427-9a3b-494d-8707-0a5735353f26">|

## Test
- Checkout branch
- Run `yarn generate`
- Ensure nothing is broken and build completes

- Run `yarn storybook`
- Ensure everything renders appropriately there (not all components are listed there yet)
